kpi/views.py

```diff
@@ -960,6 +960,12 @@ def list(self, request, *args, **kwargs):
         # Do not allow requests to retrieve more than `SUBMISSION_LIST_LIMIT`
         # submissions at one time
         limit = filters.get('limit', settings.SUBMISSION_LIST_LIMIT)
+        try:
+            limit = int(limit)
+        except ValueError:
+            raise exceptions.ValidationError(
+                {'limit': _('A valid integer is required')}
+            )
```

Comment:
I would have refactored the code to use the same validator in both place (here and `BaseDeploymentBackend`)
I'll do it in one of the PRs for `two-databases`.